tools/OMIcheck/OMIcheck.ps1

```diff
@@ -1,127 +1,460 @@
+            $omiPkgVer = $split[2]
+        }
+
+        if ($ladMdsdPkgLine -ne $null)
+        {
+            $split = $ladMdsdPkgLine.Split(" ",[System.StringSplitOptions]::RemoveEmptyEntries)
+            $ladMdsdPkgVer = $split[2]
+        }
+
+        if ($omsAgentPkgLine -ne $null)
+        {
+            $split = $omsAgentPkgLine.Split(" ",[System.StringSplitOptions]::RemoveEmptyEntries)
+            $omsAgentPkgVer = $split[2]
+        }
+
+        $ret = "" | Select-Object -Property OmiServerVer,OmiPkgVer,LadPkgVer,OmsPkgVer
+        $ret.OmiServerVer = $omiVer
+        $ret.OmiPkgVer = $omiPkgVer
+        $ret.LadPkgVer = $ladMdsdPkgVer
+        $ret.OmsPkgVer = $omsAgentPkgVer
+
+        return $ret
+    }
+
+    return $null
+}
+
+# Get the versions of the packages from a VM.
+function IsVMVulnerableOMI($VM)
 {
-    #DEBUG: Limit to one subscription only.
-    #if ($sub.Name -eq "My subscription name here")
-    #{
-        # Set Azure Subscription context    
-        Set-AzContext -Subscription $sub.Id
-
-        Write-Output "Listing Virtual Machines in subscription '$($sub.Name)'"
-        $VMs = Get-AzVM -Status
-        $VMsSorted = $VMs | Sort-Object -Property ResourceGroupName
-        $PreviousRG = ""
-        foreach($VM in $VMs)
-        {
-            #DEBUG: limit to 1 vm only
-            #if ($vm.name -ne "my-testing-vm")
-            #{
-            #    continue
-            #}
-
-            # Group VMs by resource groups.
-            if ($PreviousRG -ne $VM.ResourceGroupName)
+    # TODO: Consider setting timeout. Parameter does not exist, -AsJob is an option for v2.
+    $check = Invoke-AzVMRunCommand -ResourceGroupName $VM.ResourceGroupName -Name $VM.Name -CommandId 'RunShellScript' -ScriptPath $checkScriptPath
+
+    return ParsePkgVersions($check.Value.Message)
+}
+
+# Get the versions of the packages from a VMSS instance.
+function IsVMssVulnerableOMI($VMss, $vmssInst)
+{
+    # TODO: Consider setting timeout. Parameter does not exist, -AsJob is an option for v2.
+    $check = Invoke-AzVmssVMRunCommand -ResourceGroupName $VMss.ResourceGroupName -VMScaleSetName $VMss.Name -InstanceId $vmssInst.InstanceId -CommandId 'RunShellScript' -ScriptPath $checkScriptPath
+
+    return ParsePkgVersions($check.Value.Message)
+}
+
+# If LAD extension is installed, nudge the Azure Guest Agent to pickup the latest bits.
+function UpdateLAD($VM)
+{
+    $exts = Get-AzVMExtension -VMName $VM.Name -ResourceGroupName $VM.ResourceGroupName
+    $ladExt = $exts | where { ($_.Publisher -eq $LADpublisher -and $_.Name -eq $LADextName) }
+    if ($ladExt -ne $null)
+    {
+        if ($ladExt.ProvisioningState -eq "Succeeded")
+        {
+            # Trigger a goal state change for GA.
+            Set-AzVMExtension -VMName $VM.Name -ResourceGroupName $VM.ResourceGroupName -Location $ladExt.Location -Publisher $ladExt.Publisher -Name $ladExt.Name -ExtensionType $ladExt.ExtensionType -TypeHandlerVersion $ladExt.TypeHandlerVersion
```

Comment:
could save the response and check like
```
$response = Set-AzVMExtension ...
if ($response.IsSuccessStatusCode) ...

Reply:
along with the other `Update*` functions

Reply:
sure.